### PR TITLE
pipewire refactor

### PR DESCRIPTION
**Pull request by AndersonTorres**
## Description of changes

Use `lib.meson*` functions instead of custom code.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by AndersonTorres** [comment]
@ofborg build pipewire

File: pkgs/development/libraries/pipewire/default.nix [hunk 1]
```diff
@@ -71,8 +71,6 @@
 }:
 
 let
-  mesonEnableFeature = b: if b then "enabled" else "disabled";
-
   self = stdenv.mkDerivation rec {
```
Comment (uninsane) [inline pkgs/development/libraries/pipewire/default.nix:74]: you can drop the entire `let` scope: nothing here makes use of `self` being named.

**Comment by uninsane** [comment]
i was about to propose a very similar change (over [here](https://github.com/NixOS/nixpkgs/commit/984d72827bbae4b55442655adacb6201e8766a03)). this PR looks good though -- at a glance -- and it shows 0 rebuilds which is a good indicator of no meaningful typos. any reason it's marked draft?

**Comment by AndersonTorres** [comment]
> any reason it's marked draft?

1. A long time ago I changed a PR to draft,  and now GitHub treats it as default. Since then I liked the idea!
2. By any strange reason OfBorg refused to compile this, so I kept it as draft.

**Comment by K900** [comment]
It's probably just timing out because it's trying to build on top of staging and not enough of it is cached. I like the change, FWIW.

**Comment by AndersonTorres** [comment]
The messages from ofborg don't look like the typical timeout, tho.

**Comment by K900** [comment]
Oh wait, you mean the test failures. Yeah, that's (kinda) expected, that test is very flaky in VMs.

**Comment by AndersonTorres** [comment]
That being said, I will open for analysis.

**Pull request merged**